dropshot/src/api_description.rs

```diff
@@ -1579,6 +1588,17 @@ pub struct TagExternalDocs {
     pub url: String,
 }
 
+/**
+ * Dropshot/Progenitor features used by endpoints which are not a part of the base OpenAPI spec.
+ */
+#[derive(Copy, Clone, Debug, Default, Eq, PartialEq)]
+pub enum ExtensionMode {
+    #[default]
```

Comment:
i actually changed to this from a proper bare `impl Default` after an earlier PR comment pointed out that the rust-toolchain.toml for this project declares rust 1.62 - simple enough to revert to what i originally had if it's a concern at all

Reply:
I should have clarified. I don't know if we have an MSRV! Yes, I know we specify the toolchain; that's at least in part to ensure repeatable builds as we have several tests for the `endpoint` macros that are a bit.. finicky with regard to version dependence.

I love `#[default]`; I don't know if using it would be annoying for our internal consumers (or maybe for external consumers; 👋 I don't know if you exist, but if you do, we love you!)

Reply:
I did mention the derive Default but we don't have an MSRV I could find so depends on how conservative we wanna be here I guess.

In terms of our own internal consumers, of the ones I know of:
- propolis lists an [MSRV of 1.62](https://github.com/oxidecomputer/propolis/#minimum-supported-rust-version-msrv)
- omicron is pinned on an [old nightly](https://github.com/oxidecomputer/omicron/blob/9bbf2a0fa436542e6bfe2f403114ba280fe3730d/rust-toolchain.toml#L13) (due to unstable features): nightly-2022-04-27
- crucible is also pinned on an [even older nightly](https://github.com/oxidecomputer/crucible/blob/1eac1be81e0b3d624f2d09aecc40601a7aaef5f1/rust-toolchain.toml#L2): nightly-2021-11-24
- buildomat doesn't have a rust-toolchain or note about version
- cio is pinned on an [old nightly](https://github.com/oxidecomputer/cio/blob/209300a352cb653fac660c0234f80749b5974a02/rust-toolchain#L1): nightly-2022-03-13

Reply:
Do the old nightly's support #[default]? If so I say we go ahead and keep it. If not... I guess maybe roll them forward first?

Reply:
i think i'm not quite as inclined to have us drag everything forward for something as trivial as some very small syntax sugar (vs. say, implementing an actual *feature* in such a way that otherwise wouldn't have been possible), but i wouldn't  say no if other folks felt strongly otherwise

Reply:
Yea, I don't think that that should hold this up.